spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/subscriber/PubSubSubscriberTemplate.java

```diff
@@ -111,6 +111,7 @@ public void setMessageConverter(PubSubMessageConverter pubSubMessageConverter) {
 
 	public void setAckExecutor(Executor ackExecutor) {
 		Assert.notNull(ackExecutor, "ackExecutor can't be null.");
+		this.defaultAckExecutor.shutdown();
```

Comment:
it's shut down in destroy().

Reply:
Why wait until `destroy()`? Once another executor is set, the default one will never be used.

Reply:
This smells like a different PR :)
Make sure it does not get shut down twice, though (no harm, but confusing).